salt/modules/win_path.py

```diff
@@ -126,10 +130,16 @@ def add(path, index=0):
     if index > len(sysPath):
         index = len(sysPath)
 
-    localPath = os.environ["PATH"].split(os.pathsep)
+    pathsep = str(os.pathsep)  # future lint: disable=blacklisted-function
```

Comment:
Windows registry is Unicode and the PATH should be Unicode as well. As NTFS is Unicode filenames.
FAT based files systems do not support Unicode, but that would be unto the person to make sure the input is valid ASCII.
So the reg function read the current path will return Unicode, and you need to send Unicode back when setting it.

Reply:
It seems that `reg.set_value` would convert to unicode anyway, so I'm not sure that's too big of a concern, but I've changed the calls to it so that a unicode PATH string is passed to it.

You are wrong however, with respect to `os.environ` containing unicode data. On Python 2 this causes a traceback:

```
  (snip)
  ...
  File "c:\python27\lib\subprocess.py", line 390, in __init__
    errread, errwrite)
  File "c:\python27\lib\subprocess.py", line 640, in _execute_child
    startupinfo)
TypeError: environment can only contain strings
```

Reply:
Thanks for trying.  It just provides more proof that PY2 is no longer suitable for Windows, PY2 has too many hacks to work in modern versions of Windows which use Unicode.  Even PowerShell 6 has now moved to UTF-8 without BOM instead of UTF with BOM to be more cross platform friendly.